### PR TITLE
Add VMware transport

### PR DESCRIPTION
**Pull request by jerryaldrichiii**
This adds the VMware transport via PowerCLI.

See: https://github.com/inspec/inspec/issues/3133

### CODE REVIEW

File: lib/train/transports/vmware.rb [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+      @connection ||= Connection.new(@options)
+    end
+
+    class Connection < BaseConnection # rubocop:disable ClassLength
+      POWERSHELL_PROMPT_REGEX = /PS\s.*> $/
+
+      def initialize(options)
+        super(options)
+
+        options[:viserver] = options[:viserver] || options[:host]
```
Comment (clintoncwolfe) [inline lib/train/transports/vmware.rb]: Fun shorthand: try `||=`.  Like `+=`, but for or.  It's great for setting defaults.
```
options[:viserver] ||= options[:host]
     
```

File: lib/train/transports/vmware.rb [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+          else
+            message += result.stderr.gsub(/-Password .*\s/, '-Password REDACTED')
+          end
+
+          raise message
+        end
+      end
+
+      def local?
+        true
```
Comment (clintoncwolfe) [inline lib/train/transports/vmware.rb]: Is this constant?

File: lib/train/transports/vmware.rb [hunk 3]
```diff
@@ -0,0 +1,190 @@
+# encoding: utf-8
```
Comment (clintoncwolfe) [inline lib/train/transports/vmware.rb]: There is a lot going on here that I really wish we had integration tests for.  The unit tests are solid - I believe that the accessors and option parsers work.  I'm sure info about integration testing is coming, likely in the resource pack.

Comment (clintoncwolfe) [inline lib/train/transports/vmware.rb]: That said, pleas update the README.md to include vmware, and provide configuration information there.

File: lib/train/platforms/detect/specifications/api.rb [hunk 4]
```diff
@@ -11,6 +11,8 @@ def self.load
       plat.name('aws').in_family('cloud')
       plat.name('azure').in_family('cloud')
       plat.name('gcp').in_family('cloud')
+      plat.name('vmware').in_family('cloud')
```
[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply lib/train/platforms/detect/specifications/api.rb]: Yup, I struggled with this as well. After discussing, @jquick and @chris-rock decided that is should be under `cloud`.

File: lib/train/transports/vmware.rb (hunk 3 above)
Comment (jerryaldrichiii) [reply lib/train/transports/vmware.rb]: Yup, I too would like integration tests, I'm just not quite sure how to approach it here.

Comment (jerryaldrichiii) [reply lib/train/transports/vmware.rb]: I've added sections covering basic usage. Detailed info (such as PWSH/PowerCLI installation will be in the resource pack).

Does that work for you? 😄

File: lib/train/transports/vmware.rb (hunk 1 above)
Comment (jerryaldrichiii) [reply lib/train/transports/vmware.rb]: Indeed it is, will do!

File: lib/train/transports/vmware.rb (hunk 2 above)
Comment (jerryaldrichiii) [reply lib/train/transports/vmware.rb]: I think so, I'm trying to override: https://github.com/inspec/train/blob/d7b958c4cd1ee3b7993bd0fc05a85fb25544046c/lib/train/plugins/base_connection.rb#L76

File: test/unit/transports/vmware_test.rb [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    end
+
+    it 'uses the Local connection when Windows PowerShell is found' do
+      require 'train/transports/local'
+      Train::Transports::Local::Connection.expects(:new)
+      create_transport(
+        stub_options: {
+          powershell_binary: :powershell
+        }
+      ).connection
```
[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply test/unit/transports/vmware_test.rb]: Well, I expect it to create a `Local::Connection` if the PowerShell binary is proper PowerShell (vs `pwsh`).

So I did an `expect`, then created the connection. Is there a better way?

File: test/unit/transports/vmware_test.rb [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+    it 'raises certificate error when stderr matches regular expression' do
+      e = proc {
+        create_transport(
+          stub_options: {
+            mock_connect_result: mock_connect_result(
+              'Invalid server certificate', 1
+            )
+          }
+        ).connection
+      }.must_raise(RuntimeError)
```
[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply test/unit/transports/vmware_test.rb]: Hmmm...should I raise a different error? My plan was to ensure it raises, capture the exception in `e` then make sure the message is what I expect.

File: test/unit/transports/vmware_test.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (clintoncwolfe) [reply test/unit/transports/vmware_test.rb]: Ahh, I see.  the `expects` should fire.  Too many testing paradigms on this project :-)

**Pull request merged**